src/features/projectsV2/ProjectsMap/MapFeatureExplorer/microComponents/MapSettingsSection.tsx

```diff
@@ -18,20 +18,21 @@ type Props = BaseProps &
   );
 
 const isValidGroup = (config: LayerConfig[]): boolean =>
-  config.length > 0 && config.some((layerConfig) => layerConfig.isAvailable);
+  config.length > 0 &&
+  config.some((layerConfig) => layerConfig.isAvailable && layerConfig.canShow);
 
 const GroupedLayers = ({
   config,
   mapOptions,
   updateMapOption,
 }: { config: LayerConfig[] } & BaseProps) => {
-  const availableLayers = config.filter(
-    (layerConfig) => layerConfig.isAvailable
+  const availableLayersConfig = config.filter(
+    (layerConfig) => layerConfig.isAvailable && layerConfig.canShow
```

Comment:
`availableLayersConfig` is recalculated on every render, Wrap it in `useMemo` to optimize performance:

Reply:
Resolved in f186acc12bb0393e6c681732132d8a7fa6ab7b9c